vllm/worker/neuron_model_runner.py

```diff
@@ -233,6 +243,36 @@ def prepare_model_input(
                                    sampling_metadata=sampling_metadata,
                                    multi_modal_kwargs=multi_modal_kwargs)
 
+    def _update_neuron_generation_config(self, sampling_metadata):
+        # Update Neuron Genetation Config
+        current_generation_config = self.model_config.generation_config
+        assert current_generation_config is not None, (
+            f"Failed to update generation_config, "
+            f"current generation config is {current_generation_config}")
+
+        top_k = current_generation_config.top_k.copy()
+        top_p = current_generation_config.top_p.copy()
+        temperature = current_generation_config.temperature.copy()
+        for index, sequence_group_to_sample in enumerate(
+                sampling_metadata.seq_groups):
+            top_k[index] = sequence_group_to_sample.sampling_params.top_k
+            top_p[index] = sequence_group_to_sample.sampling_params.top_p
+            temperature[index] = \
+                sequence_group_to_sample.sampling_params.temperature
+
+        # Only call update the generation config is the new config is different.
+        # This avoids calling update_generation_config for every token within
+        # the same sequence.
+        if (top_k != current_generation_config.top_k
+                or top_p != current_generation_config.top_p
+                or temperature != current_generation_config.temperature):
+            current_generation_config.top_k = top_k
+            current_generation_config.top_p = top_p
+            current_generation_config.temperature = temperature
```

Comment:
```
top_k=[1] * self.scheduler_config.max_num_seqs,
```
top_k is initialized as a vector. Are we trying to override with a scalar value here ?

Reply:
nope, here the `top_k` is also a vector as it's a copy of the original vector `top_k = current_generation_config.top_k.copy()`